SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularitySchedulerLock.java

```diff
@@ -12,24 +13,65 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(SingularitySchedulerLock.class);
 
-  private final ReentrantLock lock;
+  private final ReentrantLock stateLock;
+  private final ReentrantLock offersLock;
+  private final ConcurrentHashMap<String, ReentrantLock> requestLocks;
 
   @Inject
   public SingularitySchedulerLock() {
-    this.lock = new ReentrantLock();
+    this.stateLock = new ReentrantLock();
+    this.offersLock = new ReentrantLock();
+    this.requestLocks = new ConcurrentHashMap<>();
   }
 
-  public long lock(String name) {
+  public long lock(String requestId, String name) {
```

Comment:
Could it be a good idea to accept the calling class here instead of a name, and then log out the class name in the trace logs below? Just to ensure that the names we pass when locking aren't arbitrary and are useful for tracing the flow of the scheduler.

Reply:
maybe even going as far as class#methodName could be useful for that logging